### DISCUSSION TITLE
Slick making multiple write calls

### DISCUSSION THREAD

**Discussion by khardee1** [slick/slick#2479] [Questions]
I am attempting to make a function that intercepts the db.run() function in Slick in order to make a query on exclusively [Write] type queries. This query is made transactionally with the secondary query, so that it has the Postgres transaction ID to correlate, but what we find is that every time a db.run() is called with an insert, it creates 3 such [Write] queries to the database, and two of them are completely identical as far as I can tell, with identical DumpInfos attached to it, but neither actually results in a real write to the database.

Has anyone seen this 1-to-3 behavior? Is this known? Is there any way I can limit it to only the "real" write to the database?

**Comment by nafg** [comment]
Can you share a self-contained reproducer?

**Reply by khardee1** [reply]
Sorry about taking so long. Due to internal code I cannot share, as well as the difficulties of extricating said code from the larger codebase, I have not yet been successful in creating a reproducer yet. I have, in the meantime, managed to pull out the function that is in question, as well as sanitize a bunch of query logs to show this behavior. In essence, it seems like the problem is stemming from certain select queries being marked as having an Effect of Write, despite being very clearly selects. 

I am still working on a reproducer, just difficult to distill this all down and make it safe for me to share externally. In the meantime, if anything is sticking out to you based on these traces, please let me know.

https://github.com/khardee1/slick-issue/tree/main

**Reply by nafg** [reply]
Would you like to set up a call?

**Reply by nafg** [reply]
It's not enough context. I can't tell how to correlate the logs with the function. But if it's too hard to make a public reproducer you can send me a private one. Or if it's too hard to make a standalone reproducer altogether then perhaps I could help on a call.
